core/server/services/settings/ensure-settings.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+const fs = require('fs-extra'),
+    path = require('path'),
+    debug = require('ghost-ignition').debug('settings:ensure-settings'),
+    _ = require('lodash'),
+    common = require('../../lib/common'),
+    knownSettings = require('./index').knownSettings,
+    config = require('../../config');
+
+/**
+ * Makes sure that all support
+ * @returns {Promise}
+ * @description Returns settings object as defined per YAML files in
+ * `/content/settings` directory.
+ */
+module.exports = function ensureSettingsFiles() {
+    return new Promise(function (resolve, reject) {
+        return _.each(knownSettings, function (setting) {
+            const fileName = `${setting}.yaml`;
+            const filePath = path.join(config.getContentPath('settings'), fileName);
+
+            return fs.access(filePath)
```

Comment:
Why are you using `fs.access`? Just curious. As far as i know this fn is for checking if you have permissions.

Reply:
Because of this sentence in the [Node API docs](https://nodejs.org/docs/latest-v8.x/api/fs.html#fs_fs_stat_path_callback):

> Using fs.stat() to check for the existence of a file before calling fs.open(), fs.readFile() or fs.writeFile() is not recommended. Instead, user code should open/read/write the file directly and handle the error raised if the file is not available.

> To check if a file exists without manipulating it afterwards, fs.access() is recommended.

Is there a better way to do so?